prow/config/jobs.go

```diff
@@ -253,9 +253,6 @@ func (br Brancher) Intersects(other Brancher) bool {
 
 // RunsAgainstChanges returns true if any of the changed input paths match the run_if_changed regex.
 func (cm RegexpChangeMatcher) RunsAgainstChanges(changes []string) bool {
-	if cm.RunIfChanged == "" {
```

Comment:
We discussed this change earlier:
https://github.com/kubernetes/test-infra/pull/10202#discussion_r236514689
An empty regex matches all strings, so I would argue that it's super confusing to have Prow's behavior be the opposite of that. We should really not be using `RunIfChanged == ""` as an equivalent to `RunIfChanged` not set.

Reply:
The field will be `""` if it is unset, so we need to treat `""` as unset. If a user explicitly sets their regex to `""` to match everything in a field called `run_if_changed` and explicitly turns off `run_always: true` to do that ... they can deal with the fallout of it because they're not making the most sound decisions :)

Reply:
pointers are nil instead of ""

re: soundness ... `run_if_changed` usage is not a sound decision. your bespoke re2 regex is not going to accurately capture changes that affect a test.